sync/class.jetpack-sync-module-themes.php

```diff
@@ -25,6 +28,61 @@ public function init_listeners( $callable ) {
 		add_action( 'jetpack_widget_reordered', $callable );
 	}
 
+	public function sync_network_allowed_themes_change( $option, $value, $old_value, $network_id ) {
+		if ( ! isset( $_REQUEST['action'] ) ||
+			( '-1' === $_REQUEST['action'] && ! isset( $_REQUEST['action2'] ) )
+		) {
+			return;
+		}
+
+		$action = -1 === intval( $_REQUEST['action'] ) ? $_REQUEST['action2'] : $_REQUEST['action'];
+
+		if ( 'disable-selected' === $action || 'disable' === $action )  {
+			$disabled_theme_names = array_keys( array_diff_key( $old_value, $value ) );
+			$disabled_themes = array();
+			foreach ( $disabled_theme_names as $name ) {
+				$theme = wp_get_theme( $name );
+				$disabled_themes[ $name ] = array(
+					'name' => $theme->get( 'Name' ),
+					'version' => $theme->get( 'Version' ),
+					'uri' => $theme->get( 'ThemeURI' ),
+				);
+			}
+
+			/**
+			 * Trigger action to alert $callable sync listener that network themes were disabled
+			 *
+			 * @since 5.0.0
+			 *
+			 * @param mixed $disabled_themes, Array of info about network disabled themes
+			 */
+			do_action( 'jetpack_network_disabled_themes', $disabled_themes );
```

Comment:
Themes can't be both changed from enabled to disabled and changed from disabled to enabled by the same user action. When themes are disabled, the Activity Log needs to know which themes were disabled. When themes are enabled, the Activity Log needs to know which themes were enabled. Why would we send themes that are unrelated to the activity/action?